madanalysis/layout/plotflow.py

```diff
@@ -110,865 +123,1162 @@ def InitializeHistoFrequency(self,ihisto):
+                        pdfset.update(
+                            {"weights": datasets[ind].weight_collection.pdfset(pdfid)}
+                        )
+                        pdfset.update({"replicas": WeightCollection()})
+                        for idx in range(1, pdfset["members"]):
+                            pdfset["replicas"] += datasets[ind].weight_collection.pdfset(
+                                pdfid + idx
+                            )
+                        break
 
+            if len(pdfset) == 0:
+                logging.getLogger("MA5").debug("No additional source of uncertainty")
+            else:
+                logging.getLogger("MA5").debug(pdfset)
 
+            # Creating a new histo
+            histoname = "y" + hist.name + "_" + str(ind)
+            outputPy.write("    # Creating weights for histo: " + histoname + "\n")
+            outputPy.write("    " + histoname + "_weights = np.array([")
+            print(hist.summary.array_full.shape)
+            current_histo = hist.summary.array * scales[ind]
+            full_histo = hist.summary.array_full * scales[ind]
+            upper_scale, lower_scale = None, None
+            upper_pdf, lower_pdf = None, None
+            scale_unc, pdf_unc = None, None
+            if len(pdfset) != 0:
+                if pdfset["weights"].has_scale:
+                    dyn_scale = datasets[ind].dynamic_scale_choice
+                    n_point_scale = datasets[ind].n_point_scale_variation
+
+                    # Get the nominal weight
+                    central_scale = pdfset["weights"].central_scale
+                    nominal_loc = (
+                        pdfset["weights"]
+                        .get_scale(
+                            dynamic=dyn_scale, muf=central_scale, mur=central_scale
+                        )
+                        .loc
+                    )
+                    current_histo = np.squeeze(full_histo[:, nominal_loc])
+
+                    scale_vars = pdfset["weights"].get_scale_vars(
+                        point=n_point_scale, dynamic=dyn_scale
+                    )
+
+                    scale_upper_loc = scale_vars[1].loc
+                    scale_lower_loc = scale_vars[0].loc
+                    upper_scale = np.abs(
+                        np.squeeze(full_histo[:, scale_upper_loc]) - current_histo
+                    )
+                    lower_scale = np.abs(
+                        np.squeeze(full_histo[:, scale_lower_loc]) - current_histo
+                    )
+                    scale_unc = np.vstack([lower_scale, upper_scale])
+
+                if len(pdfset["replicas"]) != 0:
+                    pdfvar_loc = pdfset["replicas"].loc + nominal_loc
+                    pdfvar_histo = full_histo[:, pdfvar_loc]
+
+                    ### Choose method
+                    method = "replicas"
+                    # TODO this list may need to be extended
```

Comment:
There are actually very few non Hessian sets (NNPDF + specific PDF4LHC sets). Better to define the list in this way.